src/core/meshcop/joiner.cpp

```diff
@@ -340,6 +399,8 @@ void Joiner::SendJoinerEntrustResponse(const Coap::Header &aRequestHeader,
     memset(&responseInfo.mSockAddr, 0, sizeof(responseInfo.mSockAddr));
     SuccessOrExit(error = mCoapServer.SendMessage(*message, responseInfo));
 
+    mState = kStateJoined;
```

Comment:
From what I can tell, it still waits for the `HandleTimer` before triggering the callback. Is that correct? If so, could you explain why?